synapse/config/key.py

```diff
@@ -95,9 +95,49 @@ def read_config(self, config, config_dir_path, **kwargs):
 
         # if neither trusted_key_servers nor perspectives are given, use the default.
         if "perspectives" not in config and "trusted_key_servers" not in config:
+            logger.warn(
+                """
+                Synapse requires that a list of trusted key servers are
+                specified in order to provide signing keys for other servers in
+                the federation.
+
+                This homeserver does not have a trusted key server configured in
+                homeserver.yaml and will fall back to the default of
+                'matrix.org'.
+
+                Trusted key servers should be long lived and stable which
+                makes matrix.org a good choice for many admins, but some admins may
+                wish to choose another. To suppress this warning, the admin
+                should set 'trusted_key_servers' in homeserver.yaml to their
+                desired key server and 'suppress_key_server_warning' to 'True'.
+
+                In a future release the software defined default will be
+                removed entirely and the trusted key server will be defined
+                exclusively by the value of 'trusted_key_servers'.
+                """
+            )
             key_servers = [{"server_name": "matrix.org"}]
         else:
             key_servers = config.get("trusted_key_servers", [])
+            for server in key_servers:
+                if (
+                    server["server_name"] == "matrix.org"
+                    and not suppress_key_server_warning
+                ):
+                    logger.warn(
+                        """
+                        This server is configured to use 'matrix.org' as its
+                        trusted key server via the 'trusted_key_servers' config
+                        option. 'Matrix.org' is a good choice for a key server
+                        since it is long lived, stable and trusted. However, some
+                        admins may wish to use another server for this purpose.
+
+                        To suppress this warning and continue using
+                        'matrix.org', admins should set
+                        'suppress_key_server_warning' to 'True' in
+                        homeserver.yaml.
+                        """
+                    )
 
             if not isinstance(key_servers, list):
```

Comment:
this check is a bit pointless when you've just iterated over it assuming it was a list. move the new code after it, please.